### ISSUE TITLE
Interface type with all optional properties allows assignment of primitive types.

### ISSUE THREAD

**Issue by MLoughry** [Microsoft/TypeScript#17723]
**TypeScript Version:**  2.4.0 / nightly (2.5.0-dev.201xxxxx)

**Code**

```ts
interface SomeInterface {
    type?: string;
}

let foo: SomeInterface = 'foo'; // Passes, despite mismatched type
let bar: SomeInterface = 1; // Passes, despite mismatched type
```

**Expected behavior:**
The compiler throws an error for mismatched types

**Actual behavior:**
No exception is thrown during compilation; though, curiously, the TSServer in VS Code does flag it as an error in the IDE

![image](https://user-images.githubusercontent.com/20823465/29182113-82986674-7db2-11e7-91d4-0bc2ea77d45f.png)

**Comment by RyanCavanaugh** [comment]
@sandersn ?

**Comment by Kovensky** [comment]
It being assignable is working as intended AFAIK (anything is assignable to empty interface) but in this case it's being caught by excess property check, interesting.

As an aside, I wonder if `interface SomeInterface extends object {}` would prevent primitives from being assignable to it.

**Comment by sandersn** [comment]
@MLoughry the lack of error is expected in 2.4.0. I fixed it in 2.4.1 to report the error on primitives as well. Did you check with typescript@next or 2.4.1? There, I get the error both in the editor and from compilation.

**Comment by MLoughry** [comment]
Whoops. Apparently, when I was reproing this, I was using the globally installed tsc, which was an older version. Using the proper version now works. Sorry for the false alarm!
